plugin/semantic.py

```diff
@@ -0,0 +1,173 @@
+
+    modifier_binary = [int(x) for x in reversed(bin(modifier_int)[2:])]
+    modifiers = [semantic_tokens_legends['tokenModifiers'][i] for i in range(len(modifier_binary)) if i]
+
+    # print('tokenType: ' + token_type + '-- modifiers: ',modifiers)
+    scopes = []
+    for modifier in modifiers:
+        scope = None
+        try:
+            scope = SEMANTIC_SCOPES[modifier][token_type]
+        except KeyError:
+            pass  # print('scope not defined for modifier/tokenType: ', modifier+'/'+token_type)
+
+        if scope and scope not in scopes:
+            scopes.append(scope)
+
+    if not scopes:
+        try:
+            return(SEMANTIC_SCOPES[''][token_type])
+        except KeyError:
+            pass  # print('scope not defined for tokenType: ', token_type)
+    else:
+        return ' '.join(scopes)
+
+    return ''
+
+
+def color_scheme_error() -> None:
+    print(
+        '''
+        Color Scheme :
+        Error: Scope not defined in ColorSchme
+
+        Add the following to your color scheme
+
+        "rules":
+        [
+            {
+                "scope": "meta.semantic-token.lsp",
+                "background": "#01010101"
+            }
+        ]
+
+        NOTE: most color schemes don't utilise many available SublimeText scopes,
+        this will result in subpar semantic highligthing. You should customize
+        your theme and add scopes to you liking to take full advantage of this feature.
+
+        example rules:
+        {
+            "scope": "meta.semantic.variable",
+            "foreground": "var(variable)",
+        },
+        {
+            "scope": "meta.semantic.method",
+            "foreground": "var(method)",
+        },
+
+        {
+            "scope": "meta.semantic.method.static",
+            "foreground": "var(static_method)",
+        },
+
+        syntax: meta.semantic.tokenType.modiferType
+
+        modifierTypes are optional.
+
+            '''
+        )
+
+
+def init_semantic_draw_keys(view: sublime.View, server: str, doc_highlight_style: str) -> None:
+    r = [sublime.Region(0, 0)]
+    for modifiers in SEMANTIC_SCOPES.values():
+        for scope in modifiers.values():
+            view.add_regions(scope, r)
+
+    if doc_highlight_style == 'fill':
+        view.add_regions('lsp_highlight_texts',  r)
+        view.add_regions('lsp_highlight_reads',  r)
+        view.add_regions('lsp_highlight_writes', r)
+
+        view.add_regions('lsp_highlight_textm',  r)
+        view.add_regions('lsp_highlight_readm',  r)
+        view.add_regions('lsp_highlight_writem', r)
+    for i in range(1, 5):
+        view.add_regions('lsp{}d{}{}'.format(server, 's', i), r)
+        view.add_regions('lsp{}d{}{}'.format(server, 'm', i), r)
+
+    if doc_highlight_style != 'fill':
+        view.add_regions('lsp_highlight_texts',  r)
+        view.add_regions('lsp_highlight_reads',  r)
+        view.add_regions('lsp_highlight_writes', r)
```

Comment:
- This is quite bad from maintenance point of view.
 - It's not really specific to semantic tokens
 - It can easily get out of sync with the actual implementation of those regions

I think this should be both in a different, more generic place (maybe `views.py`) and be done in a way that doesn't risk getting out of date easily. Should maybe be done using some sort of pattern where various places that create regions provide the regions they operate on and the "priority" those should be drawn.

This in fact could be done as a separate PR since it's not necessarily specific to this task.

Reply:
Also there probably would be issues with that if the user changes settings at runtime.